lib/spack/spack/modules/common.py

```diff
@@ -801,6 +803,32 @@ def verbose(self):
         return self.conf.verbose
 
 
+def ensure_modules_are_enabled_or_warn():
+    """Ensures that, if a custom configuration file is found, module file generation is enabled.
+    If a custom configuration is found when module file generation is disabled, emits a warning.
+    """
+    for scope in spack.config.config.file_scopes:
+        # Skip default configuration
+        if scope.name.startswith("default"):
+            continue
+
+        modules_yaml = pathlib.Path(scope.path) / "modules.yaml"
```

Comment:
environments?

Reply:
Done in 7add3bd The warning looks like:
![Screenshot from 2023-04-28 14-40-43](https://user-images.githubusercontent.com/4199709/235153648-9c9869ee-0c53-4d5d-9114-b96f1e23ba15.png)